cardano-api/src/Cardano/Api/IPC.hs

```diff
@@ -586,17 +592,26 @@ queryNodeLocalState :: forall mode result.
                     -> Maybe ChainPoint
                     -> QueryInMode mode result
                     -> IO (Either AcquiringFailure result)
-queryNodeLocalState connctInfo mpoint query = do
-    resultVar <- newEmptyTMVarIO
-    connectToLocalNode
-      connctInfo
+queryNodeLocalState connctInfo mpoint query =
+  runOopsInEither $ queryNodeLocalState_ connctInfo mpoint query
```

Comment:
`oops` provides combinators to make wrapping `oops` code into non-`oops` code easy.

The ease of conversion is two ways, but we only show one direction here.

Here the [`runOopsInEither`](https://hackage.haskell.org/package/oops-0.2.0.0/docs/Control-Monad-Oops.html#v:runOopsInEither) function accepts an `oops` function that can only throw one error and returns a new function that returns an `Either` with that error on the LHS.